extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -151,7 +199,7 @@ void checkTriggers() {
         }
         ZonedDateTime now = ZonedDateTime.now();
         LOG.tracef("Check triggers at %s", now);
-        for (ScheduledTask task : scheduledTasks) {
+        for (ScheduledTask task : scheduledTasks.values()) {
```

Comment:
are we sure that we do not want to preserve the insertion order here ?

Reply:
The ordering should not matter. I mean all the tasks need to be processed anyway.